diff --git a/dist/exports/AppRegistry/index.js b/dist/exports/AppRegistry/index.js
index 4f7921eb718e08992b1f4cb8eec9e23ebb3dfa46..50f104a305b74c8a165e97066c2d364e5e902192 100644
--- a/dist/exports/AppRegistry/index.js
+++ b/dist/exports/AppRegistry/index.js
@@ -12,7 +12,6 @@
 
 import _objectSpread from "@babel/runtime/helpers/objectSpread2";
 import invariant from 'fbjs/lib/invariant';
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import renderApplication, { getApplication as _getApplication } from './renderApplication';
 var emptyObject = {};
 var runnables = {};
@@ -80,6 +79,6 @@ export default class AppRegistry {
     wrapperComponentProvider = provider;
   }
   static unmountApplicationComponentAtRootTag(rootTag) {
-    unmountComponentAtNode(rootTag);
+    // Nothing
   }
 }
diff --git a/dist/exports/AppRegistry/index.js.flow b/dist/exports/AppRegistry/index.js.flow
index 147838373ce7b043eb8020e03a67c131bb139ef8..52b260ff69af6529e3c5ea106e5a66db5d50d339 100644
--- a/dist/exports/AppRegistry/index.js.flow
+++ b/dist/exports/AppRegistry/index.js.flow
@@ -13,7 +13,6 @@
 import type { Application } from './renderApplication';
 import type { ComponentType, Node } from 'react';
 import invariant from 'fbjs/lib/invariant';
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import renderApplication, { getApplication } from './renderApplication';
 type AppParams = Object;
 type Runnable = {|
diff --git a/dist/exports/render/index.js b/dist/exports/render/index.js
index aa91a2aad4d0b014735bb0efe600536715fdf6e2..ff6bc2eeb25af74b8bff61112da48efd0812d62d 100644
--- a/dist/exports/render/index.js
+++ b/dist/exports/render/index.js
@@ -9,9 +9,7 @@
 
 'use client';
 
-import { hydrate as domLegacyHydrate, render as domLegacyRender } from 'react-dom';
 import { createRoot as domCreateRoot, hydrateRoot as domHydrateRoot } from 'react-dom/client';
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import { createSheet } from '../StyleSheet/dom';
 export function hydrate(element, root) {
   createSheet(root);
@@ -23,21 +21,10 @@ export function render(element, root) {
   reactRoot.render(element);
   return reactRoot;
 }
-export function hydrateLegacy(element, root, callback) {
-  createSheet(root);
-  domLegacyHydrate(element, root, callback);
-  return {
-    unmount: function unmount() {
-      return unmountComponentAtNode(root);
-    }
-  };
+export function hydrateLegacy(element, root) {
+  return render(element, root);
 }
-export default function renderLegacy(element, root, callback) {
-  createSheet(root);
-  domLegacyRender(element, root, callback);
-  return {
-    unmount: function unmount() {
-      return unmountComponentAtNode(root);
-    }
-  };
+
+export default function renderLegacy(element, root) {
+  return render(element, root);
 }
diff --git a/dist/exports/render/index.js.flow b/dist/exports/render/index.js.flow
index 1bd771e7c2d76dc5fbb3a2296d7ccbdcb90ba8e6..c692f69a607e671dc5e6a2ac64551185d37be213 100644
--- a/dist/exports/render/index.js.flow
+++ b/dist/exports/render/index.js.flow
@@ -9,9 +9,7 @@
 
 'use client';
 
-import { hydrate as domLegacyHydrate, render as domLegacyRender } from 'react-dom';
 import { createRoot as domCreateRoot, hydrateRoot as domHydrateRoot } from 'react-dom/client';
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import { createSheet } from '../StyleSheet/dom';
 declare export function hydrate(element: any, root: any): any;
 declare export function render(element: any, root: any): any;
diff --git a/dist/exports/unmountComponentAtNode/index.js b/dist/exports/unmountComponentAtNode/index.js
deleted file mode 100644
index 925051c1c2228fd5698e14f0f3b0aef09799b102..0000000000000000000000000000000000000000
diff --git a/dist/exports/unmountComponentAtNode/index.js.flow b/dist/exports/unmountComponentAtNode/index.js.flow
deleted file mode 100644
index b950090d15c926c550757ce6632acbb1d0bddce2..0000000000000000000000000000000000000000
diff --git a/dist/index.js b/dist/index.js
index 4f699876d1902f758c3af4df5438e8ecc78bd6b4..a91c02a651b2822a010cb7a0c50dcdb3df8250b7 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2,7 +2,6 @@ export { default as unstable_createElement } from './exports/createElement';
 export { default as findNodeHandle } from './exports/findNodeHandle';
 export { default as processColor } from './exports/processColor';
 export { default as render } from './exports/render';
-export { default as unmountComponentAtNode } from './exports/unmountComponentAtNode';
 export { default as NativeModules } from './exports/NativeModules';
 
 // APIs
diff --git a/dist/index.js.flow b/dist/index.js.flow
index fec1fd60368455f9f69d1241f78e4802ed587cfe..7e33eeaec782f9fec52b1ba9598cfb86abec08f6 100644
--- a/dist/index.js.flow
+++ b/dist/index.js.flow
@@ -3,7 +3,6 @@ export { default as unstable_createElement } from './exports/createElement';
 export { default as findNodeHandle } from './exports/findNodeHandle';
 export { default as processColor } from './exports/processColor';
 export { default as render } from './exports/render';
-export { default as unmountComponentAtNode } from './exports/unmountComponentAtNode';
 export { default as NativeModules } from './exports/NativeModules';
 
 // APIs
diff --git a/src/exports/AppRegistry/index.js b/src/exports/AppRegistry/index.js
index 53a0af51a83c3ee751c93e5d6fb4d2023b84efd4..aea5aa11ecea4826f9a80502624abf6351bdef03 100644
--- a/src/exports/AppRegistry/index.js
+++ b/src/exports/AppRegistry/index.js
@@ -14,7 +14,6 @@ import type { Application } from './renderApplication';
 import type { ComponentType, Node } from 'react';
 
 import invariant from 'fbjs/lib/invariant';
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import renderApplication, { getApplication } from './renderApplication';
 
 type AppParams = Object;
@@ -146,6 +145,6 @@ export default class AppRegistry {
   }
 
   static unmountApplicationComponentAtRootTag(rootTag: Object) {
-    unmountComponentAtNode(rootTag);
+    // Nothing
   }
 }
diff --git a/src/exports/render/index.js b/src/exports/render/index.js
index c578e3e619257110927467050eb872c2d140801e..6499819ea9efa1648b9a1fc42f9394014cab2b1f 100644
--- a/src/exports/render/index.js
+++ b/src/exports/render/index.js
@@ -9,16 +9,11 @@
 
 'use client';
 
-import {
-  hydrate as domLegacyHydrate,
-  render as domLegacyRender
-} from 'react-dom';
 import {
   createRoot as domCreateRoot,
   hydrateRoot as domHydrateRoot
 } from 'react-dom/client';
 
-import unmountComponentAtNode from '../unmountComponentAtNode';
 import { createSheet } from '../StyleSheet/dom';
 
 export function hydrate(element, root) {
@@ -33,22 +28,10 @@ export function render(element, root) {
   return reactRoot;
 }
 
-export function hydrateLegacy(element, root, callback) {
-  createSheet(root);
-  domLegacyHydrate(element, root, callback);
-  return {
-    unmount: function () {
-      return unmountComponentAtNode(root);
-    }
-  };
+export function hydrateLegacy(element, root) {
+  return render(element, root);
 }
 
-export default function renderLegacy(element, root, callback) {
-  createSheet(root);
-  domLegacyRender(element, root, callback);
-  return {
-    unmount: function () {
-      return unmountComponentAtNode(root);
-    }
-  };
+export default function renderLegacy(element, root) {
+  return render(element, root);
 }
diff --git a/src/exports/unmountComponentAtNode/index.js b/src/exports/unmountComponentAtNode/index.js
deleted file mode 100644
index c3da00e1114c19f3236697fa5c97456d823bf5db..0000000000000000000000000000000000000000
diff --git a/src/index.js b/src/index.js
index f1a6f23fca1cc5971412cdf39d95384a4663bc3e..7e33eeaec782f9fec52b1ba9598cfb86abec08f6 100644
--- a/src/index.js
+++ b/src/index.js
@@ -3,7 +3,6 @@ export { default as unstable_createElement } from './exports/createElement';
 export { default as findNodeHandle } from './exports/findNodeHandle';
 export { default as processColor } from './exports/processColor';
 export { default as render } from './exports/render';
-export { default as unmountComponentAtNode } from './exports/unmountComponentAtNode';
 export { default as NativeModules } from './exports/NativeModules';
 
 // APIs